# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/parser/stencil-eager-delazify.js
# Commit: 36780658b33a
# Full Hash: 36780658b33a050e82385161eab2e5e2e5e90064
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 2: Rewrite concurrent delazification's input/output based on InitialStencilAndDelazifications. r=nbp
#   
#   The concurrent delazification internal is still using CompilationStencilMerger.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230388
# ==============================================================================

diff -r 31d3f2901db9 -r 36780658b33a js/src/jit-test/tests/parser/stencil-eager-delazify.js
--- a/js/src/jit-test/tests/parser/stencil-eager-delazify.js	Wed Dec 04 23:44:33 2024 +0000
+++ b/js/src/jit-test/tests/parser/stencil-eager-delazify.js	Wed Dec 04 23:44:33 2024 +0000
@@ -14,7 +14,7 @@
 let depth = 4;
 // Number of additional parser & delazification workload running concurrently
 // with the one we are attempting to measure, such that we can see differences
-// in the report of `isInStencilCache`.
+// in the report of `isDelazificationPopulatedFor`.
 let load = 14;
 
 // Return the number of function generated by a to `depthFirstExec` given the
@@ -28,7 +28,7 @@
 function depthFirstExec(indent, name, w, d) {
   let fun = `${indent}function ${name} (arg) {\n`;
   let inner = "";
-  let val = `arg + isInStencilCache(${name})`;
+  let val = `arg + isDelazificationPopulatedFor(${name})`;
   if (d > 0) {
     for (let i = 0; i < w; i++) {
       inner += depthFirstExec(`${indent}  `, `${name}_${i}`, w, d - 1);
@@ -58,18 +58,13 @@
 const stencil = finishOffThreadStencil(jobs[0]);
 evalStencil(stencil, options);
 
-waitForStencilCache(raceMe);
+waitForDelazificationOf(raceMe);
 let start = raceMe(0);
 let mid = raceMe(0);
 let end = raceMe(0);
 
-// The garbage collector should clear the cache and suppress all delazification
-// tasks.
-gc();
-let afterGc = raceMe(0);
-
 assertEq(1 <= start, true);
 assertEq(start <= mid, true);
 assertEq(mid <= end, true);
 assertEq(end <= count(width, depth), true);
-assertEq(afterGc, 0);
+print(start, mid, end);